### DISCUSSION TITLE
WiinUPro no longer able to add WiiU Pro controller as XInput device on Windows 10

### DISCUSSION THREAD

**Discussion by smeep0** [KeyPuncher/WiinUPro#149] [Help]
![image](https://user-images.githubusercontent.com/34761819/225296081-a356429c-e82e-4d3e-8369-94b0d25f590c.png)

Running Windows 10 22H2 and after the recent Windows update the WiiU Pro Controller could no longer be added as an XInput device.  What's worse is that unlike the issue in Windows 11, the controller seems to blink indefinitely as if it hasn't synchronized correctly. Meanwhile Windows 10 bluetooth settings shows the device as paired.

![image](https://user-images.githubusercontent.com/34761819/225297543-7995f88f-a843-40f7-814c-2ebfc9822977.png)

**Comment by KeyPuncher** [comment]
Do you normally use WiinUPro? Just asking since nothing in the screen shots look suspicious. Would choose Add under XInput Devices (which if that didn't do anything then sounds like a potential driver issue) and use Connect near the pro controller then load an XInput Profile.

**Comment by smeep0** [comment]
> Do you normally use WiinUPro? Just asking since nothing in the screen shots look suspicious. Would choose Add under XInput Devices (which if that didn't do anything then sounds like a potential driver issue) and use Connect near the pro controller then load an XInput Profile.

Yes, I used WiinUPro as a method to turning a Wii U Pro Controller into an XInput device without needing dongles like the Mayflash Wii to PC Adapter.

Clicking on the Connect button does nothing and while the program recognizes and is able to Sync with the controller using the Sync button (versus using Windows 10's Bluetooth syncing method which will prompt the user for a PIN), something in Windows prevents it from being recognized as a legitimate XInput device. This is also despite having an XInput profile ready as well.

In an older version of Windows 10, this wasn't really an issue. Now much like having WiinUPro on Windows 11, while it can recognize the controller properly as a Wii U Pro Controller, it refuses to recognize it as a legitimate XInput device like before.

Edit: Tested it again recently. This time it just won't plain "Sync" anymore (lights on the controller will keep blinking). WiinUPro will still be able to find the controller and bypass the PIN code, Manually loading a profile will also cause the program to crash.

**Reply by KeyPuncher** [reply]
Okay, good info. It may still be a while longer before I get back to this. Bluetooth has been like a blackberry bush for this project.
